src/utilities.test.js

```
@@ -395,3 +396,42 @@ describe('formatIdSpacing', () => {
     expect(formatIdSpacing(false)).toBe(null);
   });
 });
+
+describe('userHasPermissions', () => {
+  let userPermissions;
+  let permissionsToCheck;
+  beforeEach(() => {
+    userPermissions = ['a', 'b', 'c'];
+    permissionsToCheck = ['a', 'c'];
+  });
+
+  it('returns true if the user has all the needed permissions', () => {
+    expect(userHasPermissions(permissionsToCheck, userPermissions)).toBe(true);
+  });
+
+  it('returns false if the user has some of the needed permissions', () => {
+    userPermissions = ['a'];
+    expect(userHasPermissions(permissionsToCheck, userPermissions)).toBe(false);
+  });
+
+  it('returns true if the user has more than the needed permissions', () => {
+    userPermissions.push('d', 'e');
+    expect(userHasPermissions(permissionsToCheck, userPermissions)).toBe(true);
+  });
+
+  it('returns false if userPermissions are not provided', () => {
+    userPermissions = [];
```

I almost feel like we should throw an exception in this case as it might be a sign that the user's session is stale